### PR TITLE
Shell: Let's get rid of DeprecatedString [1/N]

### PR DESCRIPTION
**Pull request by alimpfard**

### CODE REVIEW

File: Userland/Libraries/LibCore/ArgsParser.cpp [hunk 1]
```diff
@@ -7,19 +7,20 @@
...
 #include <stdio.h>
 #include <string.h>
 
-static Optional<double> convert_to_double(char const* s)
+static Optional<double> convert_to_double(StringView s)
 {
+    DeprecatedString copy = s; // strtod needs a null-terminated string.
     char* p;
-    double v = strtod(s, &p);
+    double v = strtod(copy.characters(), &p);
```
Comment (ADKaster) [inline Userland/Libraries/LibCore/ArgsParser.cpp:23]: StringView::to_double()

File: Userland/Libraries/LibCore/ArgsParser.cpp [hunk 2]
```diff
@@ -75,18 +78,23 @@ bool ArgsParser::parse(int argc, char* const* argv, FailureBehavior failure_beha
...
         if (c == -1) {
             // We have reached the end.
             break;
-        } else if (c == '?') {
+        }
+
+        if (c == '?') {
             // There was an error, and getopt() has already
             // printed its error message.
+            dbgln("A");
```
Comment (ADKaster) [inline Userland/Libraries/LibCore/ArgsParser.cpp:97]: "A" ?

File: Userland/Libraries/LibCore/ArgsParser.cpp [hunk 3]
```diff
@@ -151,6 +159,7 @@ bool ArgsParser::parse(int argc, char* const* argv, FailureBehavior failure_beha
     }
 
     if (total_values_required > values_left) {
+        dbgln("B");
```
Comment (ADKaster) [inline Userland/Libraries/LibCore/ArgsParser.cpp:162]: :b:

Maybe scrub this file for stray dbgln :)

**Review by ADKaster — COMMENTED** [review]
seems mostly fine. didn't look too closely at the Shell changes though. Hopefully no stray dbgln hiding in there too :)

File: Userland/Libraries/LibCore/ArgsParser.cpp (hunk 2 above)
Comment (alimpfard) [reply Userland/Libraries/LibCore/ArgsParser.cpp:97]: AAAAAAAAAAAA

**Pull request merged**